src/ess/diffraction/powder/correction.py

```diff
@@ -44,3 +46,67 @@ def merge_calibration(*, into: sc.DataArray, calibration: sc.Dataset) -> sc.Data
         )
     out.masks['calibration'] = calibration['mask'].data
     return out
+
+
+def lorentz_correction(da: sc.DataArray) -> sc.DataArray:
+    """Perform a Lorentz correction for ToF powder diffraction data.
+
+    This function uses this definition:
+
+    .. math::
+
+        L = d^4 \\sin\\theta
+
+    where :math:`d` is d-spacing, :math:`\\theta` is half the scattering angle
+    (note the definitions in
+    https://scipp.github.io/scippneutron/user-guide/coordinate-transformations.html).
+
+    The Lorentz factor as defined here is suitable for correcting time-of-flight data
+    expressed in wavelength or d-spacing.
+    It follows the definition used by GSAS-II, see page 140 of
+    https://subversion.xray.aps.anl.gov/EXPGUI/gsas/all/GSAS%20Manual.pdf
+
+    Parameters
+    ----------
+    da:
+        Input data with coordinates ``two_theta`` and ``dspacing``.
+
+    Returns
+    -------
+    :
+        ``da`` multiplied by :math:`L`.
+        has the same dtype as ``da``.
+    """
+    # The implementation is optimized under the assumption that two_theta
+    # is small and dspacing and the data are large.
+    out = _shallow_copy(da)
+    dspacing = _event_or_bin_coord(da, 'dspacing')
+    two_theta = _event_or_bin_coord(da, 'two_theta')
+    theta = 0.5 * two_theta
+
+    d4 = dspacing.broadcast(sizes=out.sizes) ** 4
+    if out.bins is None:
+        out.data = d4.to(dtype=out.dtype, copy=False)
+        out_data = out.data
+    else:
+        out.bins.data = d4.to(dtype=out.bins.dtype)
```

Comment:
```suggestion
        out.bins.data = d4.to(dtype=out.bins.dtype, copy=False)
```
or is that not possible?